### ISSUE TITLE
How to use community snippets add and add personal snippets

### ISSUE THREAD

**Issue by practicalli-john** [rafaeldelboni/nvim-fennel-lsp-conjure-as-clojure-ide#17]
Would it be possible to add documentation on how to use snippets with the configuration?

Specifically I would like to add personal snippets that support markdown content and mkdocs extensions.  

I would also like to evaluate community snippets such as [rafamadriz/friendly-snippets](https://github.com/rafamadriz/friendly-snippets) but it is unclear how these can be added.  Simply adding friendly snippets as a plugin in plugin.fnl does not include these snippets.

I have tried adding a `snippets` directory in the `.config/nvim/` directory and copied 

In `fnl/config/plugion/cmp.fnl` several sources for autocompletion are defined, although it is unclear what these source specifically refer to for `buffer`, `vsnip` and `luasnip`
```
(def- cmp-src-menu-items
  {:buffer "buff"
   :conjure "conj"
   :nvim_lsp "lsp"
   :vsnip "vsnp"
   :luasnip "lsnp"})
```

In Luasnip it discusses [loaders](https://github.com/L3MON4D3/LuaSnip#add-snippets), so curious to understand if these loaders are used or relevant to the fennel configuration

What are the supported approaches for snippets in the current configuration, or how could it be extended to support custom snippets?

Thank you

**Comment by rafaeldelboni** [comment]
Hey John how's it going? 

I was reading the docs you send and as I understood this friendly-snippets will show as a vscode type of snippets, so in the menu items if should fall as `:vsnip` as your example, but I'm my tests here is how it is appearing as `:luasnip` show in my test config:
<img width="828" alt="image" src="https://user-images.githubusercontent.com/1683898/206923985-e75d2435-58ed-4b7c-9045-c17c53b4d4ae.png">

So what I did to configure this was the following:
First I changed the LuaSnip lines in `nvim/fnl/config/plugin.fnl`
```fennel
  ; snippets
  :L3MON4D3/LuaSnip {:requires [:rafamadriz/friendly-snippets
                                :saadparwaiz1/cmp_luasnip]
                     :mod :lua-snip}
```

And created the following file `nvim/fnl/config/plugin/lua-snip.fnl` with the following contents:
```fennel
(module config.plugin.lua-snip
  {autoload {vscode luasnip.loaders.from_vscode}})

(vscode.lazy_load)
```

**Comment by rafaeldelboni** [comment]
I did some tests and they are expanding as intended in my tests, let me know if adding the lazy_load fix this for you as well.

**Comment by practicalli-john** [comment]
Yes, the vscode json friendly snippets are working with the configuration describe above.  Thank you.

I would also like to have a `~/.config/nvim/snippets/` directory with language specific snippets that I have created for myself, e.g `~/.config/nvim/snippets/markdown.json`

I updated the `fnl/config/plugin/lua-snip.fnl` file to pass a path using `:paths` however this doent seem to load 

```
(module config.plugin.lua-snip
  {autoload {vscode luasnip.loaders.from_vscode}})

;; (vscode.lazy_load)
(vscode.lazy_load 
  {:paths ["./snippets"]})
```  

I have tried a few variations around the [original lua code](https://github.com/L3MON4D3/LuaSnip#add-snippets) but without success

```
-- load snippets from path/of/your/nvim/config/my-cool-snippets
require("luasnip.loaders.from_vscode").lazy_load({ paths = { "./my-cool-snippets" } })
```

**Comment by practicalli-john** [comment]
Looking at the `.local/state/nvim/luasnip.log` it seems I need a package.json file.

```
ERROR | vscode-loader: Tried reading package /home/practicalli/projects/practicalli/neovim-config-redux/snippets/package.json, but it does not exist
```

So for local snippets I need to create a `package.json` file that lists each file in that directory that represents a file of snippets, similar to that of the [friendly snippets package](https://github.com/rafamadriz/friendly-snippets/blob/main/package.json)

> Maybe I should create a practicalli/neovim-snippets project and add that to my plugins config :)
